src/controllers/products.js

```diff
@@ -157,3 +159,39 @@ exports.getProduct = async (req, res, next) => {
     return next(new ErrorResponse('Product not found!', 404));
   }
 };
+
+exports.sellProduct = async (req, res, next) => {
+  const { id, userId } = req.params;
+  const { notes } = req.body;
+  const orderNotes = notes || '';
+  const product = await Product.findById(id);
+  const user = await User.findOne({ _id: userId });
+  if (!user) {
+    return next(new ErrorResponse('Access denied!', 401));
+  }
+  if (!product) {
+    return next(new ErrorResponse('Product not found!', 422));
+  }
+  if (product.status === 'Sold') {
+    return next(new ErrorResponse('Product already sold!', 400));
+  }
+  const buyer = product.requestedBuyers.find(
+    (val) => val.toString() === userId
+  );
+  if (!buyer) {
+    return next(new ErrorResponse('Not eligible to buy this product!', 404));
```

Comment:
Let's give it a meaningful message like: `The selected user is not among the products' requesters`